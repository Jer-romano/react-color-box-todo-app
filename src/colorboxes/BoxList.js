import {React, useState} from 'react';
import Box from './Box';
import NewBoxForm from './NewBoxForm';
import {v4 as uuid } from 'uuid';

const BoxList = () => {
    const [boxes, setBoxes] = useState([]);

    const renderBoxes = () => {
        return (
            <div>
                {boxes.map(b => (<Box id={b.id}
                                      key={b.id}
                                      width={b.width}
                                      height={b.height}
                                      color={b.color}
                                      removeBox={removeBox} />))}
            </div>
        );
    };

    const addBox = (box) => {
        let newBox = { ...box, id: uuid() };
        setBoxes(boxes => [...boxes, newBox]);
    };

    const removeBox = (e) => {
        let id = e.target.id;

        const result = boxes.filter(b => b.id !== id);
        setBoxes(result);
    }

    return (
        <div className='BoxList'>
            <NewBoxForm addBox={addBox}/>
            {renderBoxes()}
        </div>
    );

};

export default BoxList;