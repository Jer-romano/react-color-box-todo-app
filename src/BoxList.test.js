import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import BoxList from './colorboxes/BoxList';

it("renders without crashing", function() {
    render(<BoxList/>);
});

it("matches snapshot", function() {
    const { asFragment } = render(<BoxList/>);
    expect(asFragment()).toMatchSnapshot();
});

it("can add a colored box", function() {
    const {getByLabelText} = render(<BoxList/>);

    expect(queryByText("background-color: red")).not.toBeInTheDocument();

    const heightInput = getByLabelText("Height in pixels:");
    const widthInput = getByLabelText("Width in pixels:");
    const colorInput = getByLabelText("Background color:");
    const submitBtn = queryByText("Add Box");

    fireEvent.change(heightInput, {target: {value: 100}});
    fireEvent.change(widthInput, {target: {value: 100}});
    fireEvent.change(colorInput, {target: {value: 'red'}});
    fireEvent.click(submitBtn);

    expect(queryByText("background-color: red")).toBeInTheDocument();
    
})
