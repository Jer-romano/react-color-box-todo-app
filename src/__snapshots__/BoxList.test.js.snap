// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form>
      <label
        for="height"
      >
        Height in pixels:
      </label>
      <input
        id="height"
        name="height"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width in pixels:
      </label>
      <input
        id="width"
        name="width"
        type="text"
        value=""
      />
      <label
        for="color"
      >
        Background color:
      </label>
      <input
        id="color"
        name="color"
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
